src/snmp.c

```diff
@@ -1397,6 +1397,38 @@ static int csnmp_read_table(host_definition_t *host, data_definition_t *data) {
       break;
     }
 
+    if (res->errstat != SNMP_ERR_NOERROR) {
+      if (res->errindex != 0 && res->errindex < oid_list_len) {
```

Comment:
Is `errindex` one-based? If so, the second condition should be `res->errindex <= oid_list_len`. (or `res->errindex > oid_list_len` if you take my previous comment into account).

Reply:
Yes, `errindex` is one-based. 

Example:
https://github.com/hardaker/net-snmp/blob/a7bc508a8930a484c3a666cbea4ab226d2a3aa88/apps/snmpget.c#L210